src/http/cookie.cr

```diff
@@ -230,11 +230,11 @@ module HTTP
       def parse_cookies(header, &)
         header.scan(CookieString).each do |pair|
           value = pair["value"]
-          if value.starts_with?('"')
+          if value.starts_with?('"') && value.ends_with?('"')
             # Unwrap quoted cookie value
             value = value.byte_slice(1, value.bytesize - 2)
           end
-          yield Cookie.new(pair["name"], value)
+          yield Cookie.new(pair["name"].strip, value.strip)
```

Comment:
The [RFC 6265](https://datatracker.ietf.org/doc/html/rfc6265#section-5.2) mentioned above doesn't mention quotes, It states that the name and the value should be stripped, that's it.

I don't mind any solution, but let's come to a single decision.

Reply:
I wonder how other languages/frameworks do it.

Reply:
IIUC, leading or trailing whitespace can *only* exist when the value is quoted. On unquoted values, as well as name (which is never quoted) all extra whitespace should be matched by `\s*` and thus is not part of the capture group.

So I don't think stripping makes any sense here.
If the value is quote, it's probably intended to have leading or trailing whitespace (for whatever reason 🤷).

Reply:
Reverting the change in this line doesn't break any specs. So if it was preserved, we'd need to add some test for it.
But I think it should be removed.

Reply:
I suppose `\s*` can miss some whitespace on unquoted values because a blank also matches `CookieOctet` and the quantifiers are both greedy. So we might need to strip trailing whitespace off of an unquoted `value`, unless we can fix the quantifier greediness in regex.

Reply:
I've changed the logic to strip only unquoted values.This way the following works:
- `key= value ; path=/test` should be parsed as `<HTTP::Cookie @name="key" @value="value">`
- `key=" value "; path=/test` should be parsed as `<HTTP::Cookie @name="key" @value=" value ">`

The specs for this case are also added.